extensions/html-language-features/client/src/customData.ts

```diff
@@ -26,9 +26,16 @@ export function getCustomDataSource(toDispose: Disposable[]) {
 		}
 	}));
 
+	toDispose.push(workspace.onDidChangeTextDocument(e => {
+		const path = e.document.uri.toString();
+		if (pathsInExtensions.has(path) || pathsInWorkspace.has(path)) {
+			onChange.fire();
+		}
+	}));
+
 	return {
 		get uris() {
-			return pathsInWorkspace.concat(pathsInExtensions);
```

Comment:
what's the benefit of that change?

Reply:
I converted the path path lists in to Set<string> so the check in line 31 becomes as fast as possible.
With the change, checking of a path has changed should be O(1) but returning the list has higher complexity.
If we want to have a `Set` to track the paths and avoid having to flatten it to an array in the getter `get uris()` we would need to maintain both a set and an array in parallel. One for filtering paths and one to return it. I can change that if you want to. If it makes sense also depends a bit on how often the getter is invoked, but I guess we don't know that..

Reply:
Ah, now I get it, thanks!